File: Java Programming/Competitive Programming/Coding Challenges 15/program4.java
/*
4.Write a program which accept number from user and return multiplication of all
digits.
Input : 2395
Output : 270
Input : 1018
Output : 8
Input : 9440
Output : 144
Input : 922432
Output : 864
*/

import java.util.Scanner;

class program4
{
    static int MultDigits(int iNo)
    {
        int iMult = 1;

        String str = Integer.toString(iNo);

        for (int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            if (ch >= '1' && ch <= '9')
            {
                int iDigit = ch - '0';
                iMult = iMult * iDigit;
            }
        }

        return iMult;
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter number: ");
        iValue = sobj.nextInt();

        iRet = MultDigits(iValue);

        System.out.print("Output is : " + iRet);

        sobj.close();
    }
}

File: C++ Programming/Competitive Programming/Coding Challenges 15/program4.cpp
/*
4.Write a program which accept number from user and count frequency of 4 in it.
Input : 2395
Output : 0
Input : 1018
Output : 0
Input : 9440
Output : 2
Input : 922432
Output : 1
*/

#include <iostream>
using namespace std;

int CountFour(int iNo)
{
    int iCount = 0;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    for (; iNo != 0; iNo = iNo / 10)
    {
        int iDigit = iNo % 10;

        if (iDigit == 4)
        {
            iCount++;
        }
    }

    return iCount;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number : ";
    cin >> iValue;

    iRet = CountFour(iValue);

    cout << "Frequency is : " << iRet;

    return 0;
}